### PR TITLE
Migrate router test v3

### LINKED ISSUE #2757: Migrate router tests from hardhat to forge
This is the last remaining hardhat unit test suite and we are not instrumenting coverage following https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2752

### PR DESCRIPTION
**Pull request by ottbunn**
### Description

migrate router tests from hardhat to foundry

### Drive-by changes

no

### Related issues

- Fixes #2757 

### Backward compatibility

no

### Testing

changing the test itself

### CODE REVIEW

File: solidity/test/router.t.sol [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        assertEq(address(router.mailbox()), address(mailbox));
+        vm.expectRevert(
+            bytes("Initializable: contract is already initialized")
+        );
+        router.initialize(address(igp), address(ism));
+    }
+
+    function testEnrolledMailboxAndRouter() public {
+        bytes32 sender = TypeCasts.addressToBytes32(address(1));
+        bytes32 recipient = TypeCasts.addressToBytes32(address(router));
```
Comment (yorhodes) [inline solidity/test/router.t.sol:61]: can we fuzz these?

File: solidity/test/router.t.sol [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        router.enrollRemoteRouter(origin, sender);
+        mailbox.testHandle(origin, sender, recipient, body);
+    }
+
+    function testUnenrolledMailbox() public {
+        vm.expectRevert(bytes("MailboxClient: sender not mailbox"));
+        router.handle(origin, TypeCasts.addressToBytes32(address(1)), body);
```
Comment (yorhodes) [inline solidity/test/router.t.sol:68]: ditto on fuzzing

File: solidity/test/router.t.sol [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        domains[1] = destination;
+        bytes32[] memory addresses = new bytes32[](2);
+        addresses[0] = remote;
+        addresses[1] = remote;
+        router.enrollRemoteRouters(domains, addresses);
+        assertEq(router.domains()[0], domains[0]);
+        assertEq(router.domains()[1], domains[1]);
+    }
+
+    function formatMessage(
```
Comment (yorhodes) [inline solidity/test/router.t.sol:146]: the test mailbox has a util for this

**Comment by ottbunn** [comment]
reminder 🛎️ updated

**Review by yorhodes — APPROVED** [review]
looks great, thanks for the contribution!

**Pull request merged**